### ISSUE TITLE
[Feature Request]: Support for autocompletion in Model attributes `$appends`, `$with`, `$casts`, `$touches`, `$dates`, `$fillable`, `$guarded`, `$withCount`

### ISSUE THREAD

**Issue by ralphjsmit** [laravel-idea/plugin#657]
### Feature Description

One of the most useful autocompletions for me is the database column autocompletion on Eloquent models and in factories. One useful aspect where this autocompletion is still missing is in the Eloquent properties at the top of a model.

I think that I have listed most properties below and it would be great if you could add auto-completion for these array properties:

### Getters/attributes

- `$appends`

### Column names

- `$casts` (autocompletion on the array values does already exists, e.g. `integer`, `boolean`, etc.)
- `$fillable`
- `$guarded`
- `$dates`

### Relations

- `$with`
- `$touches`
- `$withCount` (if easy/possible)

It would be awesome if you could add support for the autocompletion on these arrays. Thanks!

**Comment by adelf** [comment]
Hi, Ralph. It already works. Maybe you forgot to generate a helper code?

<img width="182" alt="image" src="https://user-images.githubusercontent.com/2818394/195565706-af5be07d-8289-4e90-9ef1-57ec118d5f6d.png">

**Comment by ralphjsmit** [comment]
Hi Adel, thanks for your reply!

Hmm, that's interesting. I always have the helper code generated, so I don't think that's the issue.

For example, see these two screencasts:

https://stream.new/v/zgjkHFsxZxWeRefDqbgBazLPKFIDedhKKZMszs9uNxo
https://stream.new/v/176nQB5cZZneFZ7NaXdMmj6l2bo85rHKDR8gwxOKXeI

I'm using PhpStorm 2022.2.3. Probably the same issue that's happening here is also the cause for #656.

**Comment by ralphjsmit** [comment]
The screencasts are in two different projects. I'm on Laravel Idea 6.0.0.222.

**Comment by adelf** [comment]
As I see, it works but doesn't complete. Could you try Ctrl-Space in the empty strings there?

**Comment by ralphjsmit** [comment]
Sure! It says "no suggestions".

<img width="572" alt="Schermafbeelding 2022-10-13 om 12 36 35" src="https://user-images.githubusercontent.com/59207045/195575037-63fa9c1f-c385-463d-9c4d-cbe3df41695c.png">

**Comment by adelf** [comment]
Could you share your `vendor/_laravel_idea` folder?

**Comment by ralphjsmit** [comment]
@adelf Do you have an email to which I can send a zip-file? My email is rjs@ralphjsmit.com, so you can also mail directly.

**Comment by adelf** [comment]
I've sent you an email. But you can use adel@laravel-idea.com

**Comment by ralphjsmit** [comment]
Emailed you!

**Comment by adelf** [comment]
Hey, Ralph. I just saw that you use your own base Model class. Is it something strange there? )

**Comment by luilliarcec** [comment]
Hi @adelf, I have the same problem. I explain my case. I am separating the project into modules, for example "Inventory", "Invoicing", "Accounting", etc... where each module can many models. It should be noted that my tables are also separated by modules, so in my models I would have something like "src/Domain/Inventory/Models/Items" in the namespace, and my table would be "invetory_items".

That to give you a bit of context. What I did was create a trait that extracts the module and puts it as a suffix to the table, so as not to be writing the name of the table in all my models.

```php
trait HasCompositeTableName
{
    /**
     * Get the table associated with the model.
     *
     * @return string
     */
    public function getTable(): string
    {
        return $this->table ?? Str::snake(
            Str::pluralStudly(
                Str::betweenFirst(static::class, 'Domain\\', '\\').class_basename($this)
            )
        );
    }
}
```

Because my table name is "magical", and doesn't follow Laravel's nomenclature either, the plugin isn't discovering my migrations and hence the properties I have.

**Comment by adelf** [comment]
Yes, Laravel Idea doesn't run PHP code, so the ide-helper package is better here. However, you can use both - ide-helper and Laravel Idea Helper code. It should work ok.

**Comment by ralphjsmit** [comment]
Hey @adelf, yes that's correct. In most projects I'm using a simple base Eloquent Model that sits 'in-between' the base Eloquent model and the models in the project.

```php
namespace Support\Models;

use Illuminate\Database\Eloquent\Model as EloquentModel;

class Model extends EloquentModel
{
    protected $guarded = [];
    
    // Any handy functions shared by all models in the project (except User-model).
}
```

**Comment by cheesegrits** [comment]
Hi, searched for this issue and landed here.

As of a week or so ago, my Laravel IDEA stopped suggesting for model attributes like $fillable.  Worked great for the last year, then just quit working entirely.  Nada.  "No suggestions".

It has also quit suggesting types for things like $casts.  There's some other places it has quit working as well, which I can't remember off the top of my head, I'll edit this comment when I run across them again.  It's the $fillable and friends which is a problem atm, as I'm working on importing a bunch of models, and you don't realize how much you rely on this plugin for those until it quits working.  :)

I'm on PHPStorm EAP 2022.3 RC2.  This plugin has been working fine through the entire current PHPStorm EAP cycle.  I have the latest IDEA plugin version (updated about an hour ago).

Nothing unusual about my models, just simple ones that extended Eloquent Model directly, usually created with this plugin.

**Comment by gorbunov** [comment]
Verified that this fields autocomplete correctly at this time. Likely already fixed.
